docs/directives/diff-date.md

```diff
@@ -0,0 +1,23 @@
+# Diff Date
+
+DIFF-DATE is a directive for taking the difference in two dates.
+
+## Syntax
+
+```
+diff-date <column1> <column2> <destColumn>
+```
+
+## Usage Notes
+
+The DIFF-DATE directive will take the difference between two Date objects, and put difference (in milliseconds)
+into the destination column.
+
+Note that this directive can only apply on two columns whose date strings have already been parsed, either using the
+PARSE-AS-DATE directive or the PARSE-AS-SIMPLE-DATE.
+
+## Examples
+
+```
+  diff-date laterDate earlierDate dateDiff
```

Comment:
The number will just be negative.
I don't think we should take the absolute value, because some records might have dates in the field field that are larger than the date in the second field, while other records might have dates that are larger in the first field. If we take the absolute value, they'll lose that information.